src/Libraries/RevitNodes/Elements/FamilyInstance.cs

```diff
@@ -321,39 +321,54 @@ public override string ToString()
             return InternalFamilyInstance.Name;
         }
 
-       #region Public Methods
+        #region Public methods
 
         /// <summary>
-        /// Sets the Euler angle of the family instance around its local Z-axis.
+        /// Set the Euler angle of the family instance around its local Z-axis.
         /// </summary>        
         /// <param name="degree">The Euler angle around Z-axis.</param>
         /// <returns>The result family instance.</returns>
         public FamilyInstance SetRotation(double degree)
         {
-           if (this == null)
-              throw new ArgumentNullException("familyInstance");
+            if (this == null)
+            {
+                throw new ArgumentNullException("familyInstance");
+            }
 
-           TransactionManager.Instance.EnsureInTransaction(Document);
+            var oldTransform = InternalGetTransform();
+            double[] oldRotationAngles;
+            TransformUtils.ExtractEularAnglesFromTransform(oldTransform, out oldRotationAngles);
+            double newRotationAngle = degree * Math.PI / 180;
+
+            if (!oldRotationAngles[0].AlmostEquals(newRotationAngle, 1.0e-6))
+            {
+                double rotateAngle = newRotationAngle - oldRotationAngles[0];
+                var axis = Line.CreateUnbound(oldTransform.Origin, oldTransform.BasisZ);
+                ElementTransformUtils.RotateElement(Document, new ElementId(Id), axis, -rotateAngle);
+            }
 
-           // Rotate the element.
-           var oldTransform = InternalFamilyInstance.GetTransform();
-           double[] oldRotationAngles;
-           TransformUtils.ExtractEularAnglesFromTransform(oldTransform, out oldRotationAngles);
+            TransactionManager.Instance.TransactionTaskDone();
 
-           Double newRotationAngle = degree * Math.PI / 180;
+            return this;
+        }
 
-           if (!oldRotationAngles[0].AlmostEquals(newRotationAngle, 1.0e-6))
-           {
-              double rotateAngle = newRotationAngle - oldRotationAngles[0];
-              var axis = Line.CreateUnbound(oldTransform.Origin, oldTransform.BasisZ);
-              ElementTransformUtils.RotateElement(Document, new ElementId(Id), axis, -rotateAngle);
-           }
+        #endregion
 
-           TransactionManager.Instance.TransactionTaskDone();
+        #region Private helper methods
 
-           return this;
+        /// <summary>
+        /// Get the transform of the internal family instance
+        /// </summary>
+        /// <returns>The internal transform</returns>
+        private Transform InternalGetTransform()
```

Comment:
:) you could have made this method internal and the method name as GetTransform(), so that other classes may use this method if required.